src/ch01_arrays_and_strings/iq01.rs

```diff
@@ -33,21 +33,73 @@ pub fn is_unique_no_ds(input: &str) -> bool {
     true
 }
 
-pub const INPUT_EXPECTED: &[(&str, bool)] = &[
-    ("abcde", true),
-    ("hello", false),
-    ("", true),
-    ("Ff", true),
-    ("noon", false),
-    ("non", false),
-    ("Non", true),
-];
+use lazy_static::lazy_static;
+
+fn gen_input_expected() -> &'static [(&'static str, bool)] {
+    let input_expected = &[
+        ("abcde", true),
+        ("hello", false),
+        ("", true),
+        ("Ff", true),
+        ("noon", false),
+        ("non", false),
+        ("Non", true),
+    ];
+
+    input_expected
+}
+
+//lazy_static! {
+//    static ref INPUT_EXPECTED: &'static [(&'static str, bool)] = {
+//        gen_input_expected()
+//    };
+//}
+
+fn gen_global_data() -> &'static [(&'static str, bool)] {
+    let input_expected = &[
+        ("item1", true),
+        ("item2", false),
+        ("item3", true),
+        ("item4", false),
+        ("item5", true),
+    ];
+
+    input_expected
+}
+lazy_static! {
+    static ref GLOBAL_DATA: &'static [(&'static str, bool)] = { gen_input_expected() };
+}
```

Comment:
TODO: remove all of this and revert to a regular CONST slice.
(These are remains of a few experiments with the `syn::visit::Visit` trait)